text/0000-rustdoc-cargo-configuration.md

```diff
@@ -0,0 +1,412 @@
+local) and there is no way to specify a description or other useful information.
+The [`feature-metadata`] RFC will provide a way to add documentation to features
+within `Cargo.toml`. `rustdoc` should present this information, but it needs a
+way to consume it from the manifest file.
+
+The second problem is that `rustdoc` has some per-crate configuration settings,
+such as relevant URLs, that are awkward to define in Rust source files using
+attributes. It is expected that there may be further configuration options in
+the future, for specifying things like:
+
+1. Resource manifests (paths to assets, such as `KaTeX` for math rendering or
+   paths to non-image files)
+2. Data required to create non-code informational pages (such as [`clap`'s
+   derive information])
+
+This RFC provides a way to solve both problems: it specifies a way that
+`rustdoc` can gain an extensible configuration, as well as a way for it to
+receive data directly from Cargo.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+Through the [`feature-metadata`] RFC, it will be possible for a crate author to
+specify documentation, stability, and deprecation status for each feature in
+`Cargo.toml`. This will be very useful information for `rustdoc` to have, but
+there would be disadvantages to `rustdoc` extracting that information directly
+from the manifest file.
+
+The proposal is to allow Cargo to handle the parsing of the manifest and simply
+reserialize the necessary portions of the data to JSON, which can then be
+consumed by `rustdoc`.
+
+The second goal of this RFC is to allow specifying `rustdoc` crate-level
+configuration data either within a new `Cargo.toml` `[tools.rustdoc]` table, or
+in a new `rustdoc.toml` file, rather than needing to be specified within the
+Rust source files itself. Initially supported keys will be kebab case versions
+of the [`rustdoc` crate-level configuration]. For example, a `Cargo.toml` file
+could contain the following:
+
+```toml
+[tools.rustdoc]
+html-logo-url = "https://example.com/logo.jpg"
+issue-tracker-base-url = "https://github.com/rust-lang/rust/issues/"
+```
+
+Perhaps most importantly, this can be specified in the workspace `Cargo.toml`
+and be used by each member crate.
+
+For projects that do not use Cargo or want separate configuration, these options
+can also be specified in a `rustdoc.toml` file using an identical schema:
+
+```toml
+# rustdoc.toml containing the same information as above
+# the only difference is that [tools.rustdoc] has become top level
+html-logo-url = "https://example.com/logo.jpg"
+issue-tracker-base-url = "https://github.com/rust-lang/rust/issues/"
+```
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+At a high level, the following changes will be necessary:
+
+1. Cargo will accept a new `[tools.rustdoc]` table that it does not validate
```

Comment:
Please dedicate a section for providing reference-level documentation of the `tools` table.  What is the purpose and scope?  Is only the `rustdoc` key allowed for now?